packages/fxa-settings/src/components/PageChangePassword/index.tsx

```diff
@@ -2,21 +2,94 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
-import { RouteComponentProps } from '@reach/router';
+import React, { ChangeEvent, useCallback, useRef, useState } from 'react';
+import { cloneDeep } from '@apollo/client/utilities';
+import { RouteComponentProps, useNavigate } from '@reach/router';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { useBooleanState } from 'fxa-react/lib/hooks';
+import { usePasswordChanger } from '../../lib/auth';
+import { cache, sessionToken } from '../../lib/cache';
+import { useAccount, Account, Session } from '../../models';
+import AlertBar from '../AlertBar';
 import FlowContainer from '../FlowContainer';
 import InputPassword from '../InputPassword';
-import LinkExternal from 'fxa-react/components/LinkExternal';
 
 // eslint-disable-next-line no-empty-pattern
 export const PageChangePassword = ({}: RouteComponentProps) => {
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [saveBtnDisabled, setSaveBtnDisabled] = useState(true);
+  const [alertText, setAlertText] = useState<string>();
+  const [currentPasswordErrorText, setCurrentPasswordErrorText] = useState<
+    string
+  >();
+  const oldPasswordRef = useRef<HTMLInputElement>(null);
+  const newPasswordRef = useRef<HTMLInputElement>(null);
+  const confirmPasswordRef = useRef<HTMLInputElement>(null);
+  const { primaryEmail } = useAccount();
+  const navigate = useNavigate();
+  const changePassword = usePasswordChanger({
+    onSuccess: (response) => {
+      changePassword.reset();
+      sessionToken(response.sessionToken);
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const account = cloneDeep(existing);
+            account.passwordCreated = response.authAt * 1000;
+            return account;
+          },
+          session: (existing: Session) => {
+            const session = cloneDeep(existing);
+            session.verified = response.verified;
+            return session;
+          },
+        },
+      });
```

Comment:
I debated whether to do this in `auth.ts` but felt like keeping `usePasswordChanger` as simple as possible was the better approach. Having it here mirrors how we handle `useMutation` in other components.